libs/wire-api/src/Wire/API/Arbitrary.hs

```diff
@@ -0,0 +1,558 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2020 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Wire.API.Arbitrary where
+
+import Control.Lens hiding (elements)
+import qualified Data.ByteString.Char8 as BS
+import Data.Code (Timeout (Timeout))
+import Data.Currency
+import Data.IP
+import Data.Json.Util (UTCTimeMillis (..), toUTCTimeMillis)
+import Data.LanguageCodes
+import Data.List.Extra (nubOn)
+import Data.Misc
+import Data.PEM (pemParseBS)
+import Data.Proxy
+import Data.Range
+import qualified Data.Set as Set
+import qualified Data.Text as ST
+import Data.Text.Ascii
+import Data.Text.Encoding (decodeUtf8)
+import Data.UUID (nil)
+import GHC.TypeLits
+import Imports
+import qualified System.Random
+import Test.QuickCheck
+import Test.QuickCheck.Instances ()
+import Text.Hostname
+import URI.ByteString.QQ (uri)
+import Wire.API.Call.TURN
+import Wire.API.Conversation.Code as Code
+import Wire.API.CustomBackend
+import Wire.API.Provider.Service
+import Wire.API.Team
+import Wire.API.Team.Feature
+import Wire.API.Team.Invitation
+import Wire.API.Team.LegalHold
+import Wire.API.Team.LegalHold.External
+import Wire.API.Team.Member
+import Wire.API.Team.Permission
+import Wire.API.Team.Role
+import Wire.API.Team.SearchVisibility
+import Wire.API.User
+import Wire.API.User.Activation
+import Wire.API.User.Auth
+import Wire.API.User.Client.Prekey
+import Wire.API.User.Handle
+import Wire.API.User.Identity
+import Wire.API.User.Password
+import Wire.API.User.Profile
+import Wire.API.User.RichInfo
+
+newtype Octet = Octet {octet :: Word16}
```

Comment:
I've seen this when working on `brig-types`, but is this even used anywhere?

```
57 (0) develop 13:08:39 mf@lag:~/src/wire-server
$ git grep -Hn Octet
libs/brig-types/src/Brig/Types/Test/Arbitrary.hs:73:newtype Octet = Octet {octet :: Word16}
libs/brig-types/src/Brig/Types/Test/Arbitrary.hs:76:instance Arbitrary Octet where
libs/brig-types/src/Brig/Types/Test/Arbitrary.hs:77:  arbitrary = Octet <$> arbitrary `suchThat` (< 256)
tools/bonanza/test/unit/Test/Bonanza/Arbitrary.hs:176:newtype Octet = Octet {octet :: Word16}
tools/bonanza/test/unit/Test/Bonanza/Arbitrary.hs:179:instance Arbitrary Octet where
```

Reply:
It was used in the instance below, but I just inlined it.